### PR TITLE
Allow default transformers overwrite

### PR DESCRIPTION
**Pull request by felixjung**
I'm trying to migrate one space to another. This migration will require quite a lot of transformations. Since I plan to use this project as a submodule without modifying any of the existing code, I wanted a way to pass in the transformers for my content as a parameter.

What do you think?

### CODE REVIEW

File: lib/transform/transform-space.js [hunk 1]
```diff
@@ -1,15 +1,22 @@
 import Promise from 'bluebird'
-import { omit } from 'lodash/object'
-import * as transformers from './transformers'
+import { omit, defaults } from 'lodash/object'
+import * as defaultTransformers from './transformers'
+
+export const spaceEntities = [
+  'contentTypes', 'entries', 'assets', 'locales', 'webhooks'
+];
```
Comment (Khaledgarbaya) [inline lib/transform/transform-space.js]: Can you remove this `;` because we are using the standard style

File: lib/transform/transform-space.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
 import Promise from 'bluebird'
-import { omit } from 'lodash/object'
-import * as transformers from './transformers'
+import { omit, defaults } from 'lodash/object'
+import * as defaultTransformers from './transformers'
+
+export const spaceEntities = [
```
Comment (Khaledgarbaya) [inline lib/transform/transform-space.js]: I don't think it is necessary to export spaceEntities here

**Comment by Khaledgarbaya** [comment]
@felixjung Thanks a lot for the contribution, we were planing to add ability to pass a custom transformer. I made few comment there but I need to test this with the tools we have [contentful-export](https://github.com/contentful/contentful-export) [contentful-import](https://github.com/contentful/contentful-import) and [contentful-space-sync](https://github.com/contentful/contentful-space-sync)

**Comment by felixjung** [comment]
I removed the export and semi-colon. Missed the semi-colon on this one.

File: lib/transform/transform-space.js [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
  */
-export default function (space, destinationSpace) {
+export default function (
+  space, destinationSpace, customTransformers, entities = spaceEntities
+) {
+  const transformers = defaults(customTransformers, defaultTransformers)
   // TODO maybe we don't need promises here at all
-  const newSpace = omit(space, 'contentTypes', 'entries', 'assets', 'locales', 'webhooks')
-  return Promise.reduce(['contentTypes', 'entries', 'assets', 'locales', 'webhooks'], (newSpace, type) => {
+  const newSpace = omit(space, ...entities);
```
Comment (Khaledgarbaya) [inline lib/transform/transform-space.js]: still one here 😔

**Comment by Khaledgarbaya** [comment]
almost there `transform-space.js:18` still have an extra semicolon

File: lib/transform/transform-space.js (hunk 3 above)
Comment (felixjung) [inline lib/transform/transform-space.js]: Jesus... I'm bad.

**Comment by felixjung** [comment]
Don't watch TV an fix change requests 😂

**Comment by Khaledgarbaya** [comment]
Great, Thanks a lot, I ask you for a bit of patient while I test this with the other tools.
Have a great night and enjoy the Tv show you are watching 😂

**Comment by felixjung** [comment]
Sure, I'll be patient. Can use my fork in the meantime. Good night!

**Pull request merged**